cpu/qn908x/periph/spi.c

```diff
@@ -0,0 +1,341 @@
+    conf->dev->CFG &= ~SPI_CFG_ENABLE_MASK;
+
+    gpio_init(conf->copi_pin, GPIO_IN);
+    gpio_init(conf->cipo_pin, GPIO_IN);
+    gpio_init(conf->clk_pin, GPIO_IN);
+}
+#endif /* MODULE_PERIPH_SPI_RECONFIGURE */
+
+int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
+{
+    const spi_conf_t *const conf = &spi_config[bus];
+
+    mutex_lock(&locks[bus]);
+
+    /* Set SPI clock speed. This silently chooses the closest frequency, no
+     * matter how far it is from the requested one. */
+    _spi_controller_set_speed(conf->dev, clk);
+
+    if ((mode & ~(SPI_CFG_CPHA_MASK | SPI_CFG_CPOL_MASK)) != 0) {
+        return SPI_NOMODE;
+    }
+
+    DEBUG("[spi] acquire: mode CPHA=%d CPOL=%d, cs=0x%" PRIx32 "\n",
+          !!(mode & SPI_CFG_CPHA_MASK), !!(mode & SPI_CFG_CPOL_MASK),
+          (uint32_t)cs);
+
+    conf->dev->CFG =
+        (conf->dev->CFG & ~(SPI_CFG_CPHA_MASK | SPI_CFG_CPOL_MASK)) | mode;
+
+    return SPI_OK;
+}
+
+void spi_release(spi_t bus)
+{
+    assert(bus < SPI_NUMOF);
+    DEBUG("[spi] release\n");
+    mutex_unlock(&locks[bus]);
+}
+
+/**
+ * @brief: Wait for the FIFO to be empty.
+ */
+static void _spi_wait_txempty(SPI_Type *spi_bus)
+{
+    while (!(spi_bus->FIFOSTAT & SPI_FIFOSTAT_TXEMPTY_MASK)) {}
+}
+
+/**
+ * @brief Bitmask for the FIFOWR register with all the HWCS deasserted.
+ */
+#define SPI_HWCS_DEASSERT_ALL \
+    (((1u << SPI_HWCS_NUMOF) - 1) << SPI_FIFOWR_TXSSEL0_N_SHIFT)
+
+/**
+ * @brief Initialize a SPI transfer given the transfer parameters.
+ */
+static void _spi_config_transfer(spi_pending_transfer_t *tr, spi_cs_t cs,
+                                 bool cont, const void *out, void *in,
+                                 size_t len)
+{
+    tr->in = in;
+    tr->in_len = in ? len : 0;
+    tr->out = out;
+    tr->tr_len = len;
+    tr->tx_mask = SPI_HWCS_DEASSERT_ALL;
+    if (GPIO_T_IS_HWCS(cs)) {
+        /* Flag that the TX should assert this HWCS by clearing the bit. */
+        tr->tx_mask &= ~(1u << (SPI_FIFOWR_TXSSEL0_N_SHIFT + GPIO_T_HWCS(cs)));
+        if (!cont) {
+            /* Flag the End of Transfer (EOT) in the mask. This will only be
+             * used in the last byte. */
+            tr->tx_mask |= SPI_FIFOWR_EOT_MASK;
+        }
+    }
+    if (!in) {
+        /* Ignores the RX side when the @p in is NULL so we don't need to read
+         * the FIFO at all. */
+        tr->tx_mask |= SPI_FIFOWR_RXIGNORE_MASK;
+    }
+    tr->tx_mask |= SPI_FIFOWR_LEN(7);  /* Data transfers of 8 bits. */
+}
+
+/**
+ * @brief Perform a blocking SPI transfer.
+ */
+static void _spi_transfer_blocking(spi_t bus, spi_pending_transfer_t *tr)
+{
+    SPI_Type *const spi_bus = spi_config[bus].dev;
+
+    /* Configure to use the RX and TX fifo, and empty them. */
+    spi_bus->FIFOCFG = SPI_FIFOCFG_ENABLETX_MASK |
+                       SPI_FIFOCFG_ENABLERX_MASK |
+                       SPI_FIFOCFG_EMPTYTX_MASK | SPI_FIFOCFG_EMPTYRX_MASK;
```

Comment:
How about

```suggestion
    spi_bus->FIFOCFG = SPI_FIFOCFG_ENABLETX_MASK
                     | SPI_FIFOCFG_ENABLERX_MASK
                     | SPI_FIFOCFG_EMPTYTX_MASK | SPI_FIFOCFG_EMPTYRX_MASK;
```

Reply:
ok, but uncrustify likes the `|` to be two spaces more than in your example with the uncrutify-riot.cfg (so not aligned with the `=`  ).

Reply:
`uncrutify` has bugs sometimes… I'd prefer what's pretty over what `uncrustify` thinks, but this is up to personal taste anyway.